privacyidea/lib/privacyideaserver.py

```diff
@@ -158,15 +159,9 @@ def add_privacyideaserver(identifier, url, tls=True, description=""):
 @log_with(log)
 def delete_privacyideaserver(identifier):
     """
-    Delete the given server from the database
+    Delete the given server from the database.
+    Raise ResourceNotFoundError if it couldn't be found.
     :param identifier: The identifier/name of the server
     :return: The ID of the database entry, that was deleted
     """
-    ret = -1
-    pi = PrivacyIDEAServerDB.query.filter(PrivacyIDEAServerDB.identifier ==
-                                              identifier).first()
-    if pi:
-        pi.delete()
-        ret = pi.id
-    return ret
-
+    return fetch_one_resource(PrivacyIDEAServerDB, identifier=identifier).delete()
```

Comment:
I am currently not sure - does ``.delete()`` return the ID of the deleted object?

Reply:
I forgot about this too, but ``.delete()`` is actually our own ``MethodsMixin`` method which returns the ID of the deleted object :-)
https://github.com/privacyidea/privacyidea/blob/e4e736a4442e1f88f391a8f639b6d7497b9ed6bf/privacyidea/models.py#L87

Reply:
Oh. This is something I added right at the beginning - so I would have had to know about this! :-/